### PR TITLE
[extractor/rai] various fixes and improvements

### PR DESCRIPTION
**Pull request by nixxo**
- [rai] check DRM directly in the manifest because sometimes is not present in the page
- [rainews] using _EMBED_REGEX and _VALID_URL fix
- [rainews] added TEST url for iframe and DRM in manifest but not in page
- [raisudtirol] using different url for TEST because the old one was a changing weekly
- [raisudtirol] using 'formats' instead of direct 'url' to provide media info to the user
- [general] re-checked all TESTS in various classes and fixed ploblems

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)

### CODE REVIEW

File: yt_dlp/extractor/rai.py [hunk 1]
```diff
@@ -51,6 +51,10 @@ def _extract_relinker_info(self, relinker_url, video_id, audio_only=False):
                 query={'output': 45, 'pl': platform},
                 headers=self.geo_verification_headers())
 
+            if xpath_text(relinker, './license_url', default='{}') != '{}':
+                self.report_drm(video_id)
+                continue
```
Comment (pukkandan) [inline yt_dlp/extractor/rai.py:56]: Error should be raised only if ALL formats have DRM, correct?

File: yt_dlp/extractor/rai.py [hunk 2]
```diff
@@ -707,7 +719,8 @@ def _real_extract(self, url):
 
 
 class RaiNewsIE(RaiIE):
-    _VALID_URL = rf'https?://(www\.)?rainews\.it/[^?#]+-(?P<id>{RaiBaseIE._UUID_RE})(?:-[^/?#]+)?\.html'
+    _VALID_URL = rf'https?://(www\.)?rainews\.it/(?!articoli)[^?#]+-(?P<id>{RaiBaseIE._UUID_RE})(?:-[^/?#]+)?\.html'
+    _EMBED_REGEX = [r'<iframe[^>]+data-src="(?P<url>/iframe/[^?#]+?\.html)']
```
Comment (pukkandan) [inline yt_dlp/extractor/rai.py:723]: this'll have to test too many iframes. Add the domain to the regex

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/rai.py (hunk 1 above)
Comment (nixxo) [reply yt_dlp/extractor/rai.py:56]: DRM is detected and, like geoprotection, is assumed is the same for all formats.

'continue' is a leftover from testing, I'll remove it.

File: yt_dlp/extractor/rai.py (hunk 2 above)
Comment (nixxo) [reply yt_dlp/extractor/rai.py:723]: the html code for the iframe is this:
`<iframe class="lazyload" allowfullscreen="true" webkitallowfullscreen="true" mozallowfullscreen="true" allow="encrypted-media *" frameBorder="0" scrolling="no" data-src="/iframe/video/2022/07/euro2022-europei-calcio-femminile-islanda-francia-gol-video-db4bee18-de2c-4ec7-b30a-83c302a43085.html"></iframe>`

as you can see, there is no domain so... maybe I can improve the regex to make it more specific in this way:
`_EMBED_REGEX = [r'<iframe[^>]+data-src="(?P<url>/iframe/[^?#]+?{RaiBaseIE._UUID_RE}\.html)']`

Comment (nixxo) [reply yt_dlp/extractor/rai.py]: ok, done.

It should be more specific.

**Pull request merged**